core/src/main/java/org/infinispan/configuration/cache/CacheMode.java

```diff
@@ -5,6 +5,11 @@
  */
 public enum CacheMode {
    /**
+    * Data is not replicated and certain features are not possible
```

Comment:
It's not about Hibernate needs, it's about rewrapping `DataContainer` as cache. So it's more like 'what we can get without too much effort from technical perspective'.
Btw. Hibernate does not need listeners, nor statistics, I've added these to fill in the contract as closely as possible.